### DISCUSSION TITLE
How can I use `AnyRouter` in a generic way, but still have _some_ types, instead of all `any`?

### DISCUSSION THREAD

**Discussion by moltar** [trpc/trpc#4093] [Q&A] [answered]
Essentially, I want a generic AnyRouter for a function that accepts a router, but inside the fn impl I want to be able to inspect router config. I want to be able to list procedures and their types.

Right now, if I pass <R extends AnyRouter> as a generic, then all values of _def are any. I wish it was something like `Record<string, ProecedureConfig>`  or some such. 

![image](https://user-images.githubusercontent.com/491247/227983531-f37214c1-fbc6-4123-805d-a0c8b64724c1.png)

In the actual router implementation, `_def.procedures` is an object, rather than any.

![image](https://user-images.githubusercontent.com/491247/227983748-ca2eaa04-5599-46e7-9824-49c03d27cc8a.png)

To be clear, I understand that I cannot infer the actual router impl inside the fn with a generic type, but I wish there was just some resemblance of the procedure type, e.g. just having a generic `Record<string, Config>`

Also asked on Discord: https://discord.com/channels/867764511159091230/1088066576303005786

**Comment by dodas** [comment]
hey @moltar!
have you found an answer to this?

**Reply by moltar** [reply]
Nope

**Comment by KATT** [comment]
Yeah, I want this too sometimes, could you do a feature request?

**Reply by KATT** [reply, marked as answer]
Added #4257 to track this
